File: 1694-reformat-phone-number/1694-reformat-phone-number.cpp
int init=[]{ios_base::sync_with_stdio(false);cin.tie(0);cout.tie(0);return 0;}();
class Solution {
public:
    string reformatNumber(string number) {
        string nums;
        for (char c : number) {
            if (isdigit(c)) {
                nums += c;
            }
        }

        string result;
        int n = nums.size();

        int i = 0;
        while (n > 4) {
            result += nums.substr(i, 3) + "-";
            i += 3;
            n -= 3;
        }

        if (n == 4) {
            result += nums.substr(i, 2) + "-" + nums.substr(i + 2);
        } else {
            result += nums.substr(i);
        }

        return result;
    }
};

File: 1694-reformat-phone-number/1694-reformat-phone-number.rs
impl Solution {
    pub fn reformat_number(number: String) -> String {
        let cleaned_nums: String = number.chars().filter(|c| c.is_digit(10)).collect();
        
        let mut result = String::new();
        let mut i = 0;
        let mut n = cleaned_nums.len();
        
        while n > 4 {
            result.push_str(&cleaned_nums[i..i+3]);
            result.push('-');
            i += 3;
            n -= 3;
        }
        
        if n == 4 {
            result.push_str(&cleaned_nums[i..i+2]);
            result.push('-');
            result.push_str(&cleaned_nums[i+2..]);
        } else {
            result.push_str(&cleaned_nums[i..]);
        }
        
        result
    }
}